### PR TITLE
Fixes dimension table infinite scroll bug

### LINKED ISSUE #2233: Horizontal scrolling of dimension detail table broken
**Describe the bug**
The dimension detail table can have width exceeding the space available, which results in scrolling of the whole main pane.

https://user-images.githubusercontent.com/2380975/234410211-300b6140-544c-483b-9d6d-318eb142f2d0.mp4

**Expected behavior**
The dimension table area should scroll by itself. 

Moreover, entering the dimension detail table indicates that the user wants to focus on the tabular content, so the plots should probably collapse to their narrower width if the table needs more space than is available/

### PR DESCRIPTION
**Pull request by bcolloran**
Fix #2233

### CODE REVIEW

File: web-common/src/features/dashboards/dimension-table/DimensionTable.svelte [hunk 1]
```diff
@@ -214,10 +214,7 @@ TableCells – the cell contents.
   }
 </script>
 
-<div
-  bind:clientWidth={containerWidth}
-  style:height="calc(100vh - var(--header, 130px) - 8rem)"
```
Comment (bcolloran) [inline web-common/src/features/dashboards/dimension-table/DimensionTable.svelte:219]: note this `--header` css var is not defined anywhere

File: web-common/src/features/dashboards/workspace/DashboardContainer.svelte [hunk 2]
```diff
@@ -1,64 +1,72 @@
...
+    : `calc(${$navigationPaddingTween * 24}px + 1.25rem)`;
 </script>
 
-<section use:listenToNodeResize class="grid items-stretch surface">
+<section use:dashboardContainerNodeWatcher class="flex flex-col gap-y-1">
   <div
     class="explore-header border-b mb-3"
     style:padding-left={leftSide}
-    style:width={exploreContainerWidth + "px"}
+    style:width={"100%"}
```
Comment (bcolloran) [inline web-common/src/features/dashboards/workspace/DashboardContainer.svelte:34]: was able to eliminate the need to set this with JS just by setting it to be full width of the container

**Comment by bcolloran** [comment]
@hamilton -- I did a bit of cleanup on this to make it easier to review / pick up. Not a ton of changes, just enough clean up to make the dimension table work.

Unfortunately, I discovered bug #2269 when I merged main into this branch, so at the moment the main thing that this branch is designed to fix (layout of dimension table) is not available to be QAed, b.c the dimension tables are not working. Once #2269 is fixed, it can be merged into this branch, at which point I think the dimension table layout should be working again. However, a careful QA pass will be needed b/c
1) because the bug introduced in PR #2257 blocks access to the dimension table, I haven't been able to QA that no other layout bugs that were introduced in other PRs when I merged main
2) I haven't yet had time to set up the cloud deployment workflow, so I'm not sure about any layout implications that might have.

File: web-common/src/features/dashboards/dimension-table/DimensionDisplay.svelte [hunk 3]
```diff
@@ -377,28 +376,32 @@
...
-    {#if values && columns.length}
-      <DimensionTable
-        on:select-item={(event) => onSelectItem(event)}
-        on:sort={(event) => onSortByColumn(event)}
-        dimensionName={dimension?.name}
-        {columns}
-        {selectedValues}
-        rows={values}
-        {sortByColumn}
+  <div class="h-full flex flex-col" style:min-width="365px">
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/dimension-table/DimensionDisplay.svelte:379]: All these style directives can be more concise with Tailwind, e.g. `min-w-[365px]`

File: web-common/src/features/dashboards/get-element-size.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Get the total size of an element, including margins
+ * @param element
+ */
+export function getEltSize(element: HTMLElement, direction: "x" | "y") {
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/get-element-size.ts:5]: `getElmtSize` or `getElementSize` would be easier to read. I typically read "ELT" as extract-load-transform.

File: web-common/src/features/dashboards/workspace/DashboardContainer.svelte [hunk 5]
```diff
@@ -1,64 +1,70 @@
...
   .explore-metrics {
-    grid-area: metrics;
-    overflow-y: auto;
+    overflow-y: scroll;
   }
+
   .explore-leaderboards {
-    grid-area: leaderboards;
+    overflow-y: hidden;
   }
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/workspace/DashboardContainer.svelte:69]: These could all be Tailwind classes

**Review by ericpgreen2 — APPROVED** [review]
My comments were nits, so I’ll approve. Works on Cloud too. Thanks!

**Pull request merged**